features/facebookImport/src/model/analyses/activities-analysis.js

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import i18n from "../../i18n.js";
+import RootAnalysis from "./root-analysis.js";
+
+import ActivitiesMinistory from "../../components/activitiesMinistory/activitiesMinistory.jsx";
+
+export default class ActivitiesAnalysis extends RootAnalysis {
+    get title() {
+        return i18n.t("activitiesMinistory:title");
+    }
+
+    async analyze({ facebookAccount }) {
+        const activities = {
+            timestamp: [
+                ...facebookAccount.followedPages,
+                ...facebookAccount.friends,
+                ...facebookAccount.interactedAdvertisers,
+                ...facebookAccount.likedPages,
+                ...facebookAccount.receivedFriendRequests,
+                ...facebookAccount.recommendedPages,
+                ...facebookAccount.searches,
+                ...facebookAccount.unfollowedPages,
+            ],
+            timestamp_ms: [],
+        };
+
+        //for nested structures
+        facebookAccount.offFacebookCompanies.forEach((company) =>
+            activities.timestamp.push(...company.events)
+        );
+
+        facebookAccount.messageThreads.forEach((thread) =>
+            activities.timestamp_ms.push(...thread.messages)
+        );
+
+        let total = { total: 0 };
```

Comment:
We could rename `total` to `groupedActivities` maybe to avoid something like `total.total`.